test/test_integration.py

```diff
@@ -0,0 +1,114 @@
+# coding: utf-8
+
+#
+# Copyright 2014-2017 Groupon, Inc.
+# Copyright 2014-2017 The Billing Project, LLC
+#
+# The Billing Project, LLC licenses this file to you under the Apache License, version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at:
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+"""
+    Kill Bill
+
+    Kill Bill is an open-source billing and payments platform  # noqa: E501
+
+    OpenAPI spec version: 1.0.0
+    
+    Generated by: https://github.com/swagger-api/swagger-codegen.git
+"""
+
+
+from __future__ import absolute_import
+
+import unittest
+
+import killbill
+from killbill.models.account import Account  # noqa: E501
+from killbill.models.tenant import Tenant  # noqa: E501
+from killbill.models.payment_method import PaymentMethod  # noqa: E501
+from killbill.models.subscription import Subscription  # noqa: E501
+from killbill.models.invoice_dry_run import InvoiceDryRun  # noqa: E501
+from killbill.rest import ApiException
+
+
+
+class TestIntegration(unittest.TestCase):
+    """Account unit test stubs"""
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_integration(self):
+
+        # Create tenant
+        # TODO WORKING WITH ISSUES*! [ValueError: Invalid value for `api_key`, must not be `None`]
+        tenant = killbill.api.TenantApi()
+        body = Tenant(api_key='bob_5', api_secret='lazar_5')
+        tenant.create_tenant(body, 'test')
+
+        # Get tenant
+        # TODO NOT WORKING! [ApiException: (401) Reason: Unauthorized] or [ValueError: Invalid value for `api_secret`, must not be `None`]
+        body = Tenant(api_key='bob')
+        tenant.get_tenant_by_api_key(body)
+
+        # Upload XML catalog/Fetch XML catalog
+        # TODO NOT WORKING! [ApiException: (404) Reason: Not Found]
+        catalog = killbill.api.CatalogApi()
+        xml_catalog = open("/home/jfrodriguez/Documents/KillBill-Docs/SpyCarBasic.xml", "r+").read()
+        catalog.upload_catalog_xml(xml_catalog, 'test', 'bob', 'lazar')
+
+        # Create account
+        # WORKING!
+        account = killbill.api.AccountApi()
+        body = Account(name='Johny', currency='USD', state='CA', country='USA')
+        account.create_account(body, 'test', 'bob', 'lazar')
+
+        # Get account
+        # TODO NOT WORKING! [AttributeError: 'module' object has no attribute 'Str']
```

Comment:
It looks like deserialization of models doesn't work at all. A work around seems to be to add these camelized primitive types:

```
--- a/killbill/api_client.py
+++ b/killbill/api_client.py
@@ -51,10 +51,13 @@ class ApiClient(object):
     PRIMITIVE_TYPES = (float, bool, bytes, six.text_type) + six.integer_types
     NATIVE_TYPES_MAPPING = {
         'int': int,
+        'Int': int,
         'long': int if six.PY3 else long,  # noqa: F821
         'float': float,
         'str': str,
+        'Str': str,
         'bool': bool,
+        'Bool': bool,
         'date': datetime.date,
         'datetime': datetime.datetime,
         'object': object,
```

but it still fails for more complex types (e.g. `List[AuditLog]`). 😕

Reply:
This seems to do the trick:

```
diff --git a/killbill/api_client.py b/killbill/api_client.py
index d096916..7e20a9d 100644
--- a/killbill/api_client.py
+++ b/killbill/api_client.py
@@ -51,12 +51,16 @@ class ApiClient(object):
     PRIMITIVE_TYPES = (float, bool, bytes, six.text_type) + six.integer_types
     NATIVE_TYPES_MAPPING = {
         'int': int,
+        'Int': int,
         'long': int if six.PY3 else long,  # noqa: F821
         'float': float,
         'str': str,
+        'Str': str,
         'bool': bool,
+        'Bool': bool,
         'date': datetime.date,
         'datetime': datetime.datetime,
+        'Datetime': datetime.datetime,
         'object': object,
     }

         if data is None:
             return None
-
         if type(klass) == str:
-            if klass.startswith('list['):
-                sub_kls = re.match('list\[(.*)\]', klass).group(1)
+            if klass.startswith('list[') or klass.startswith('List['):
+                sub_kls = re.match('[lL]ist\[(.*)\]', klass).group(1)
                 return [self.__deserialize(sub_data, sub_kls)
                         for sub_data in data]
```

Not quite sure why it's needed though :/ Maybe you could make these changes in the generator,  re-generate `api_client` and see if it fixes some tests?

Reply:
@pierre It works! Thanks for this solution. I already updated the python generator.